src/pages/ProductDetail/index.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import ProductGalleryGrid from '../../components/ProductGalleryGrid';
+import StyledButton from '../../components/StyledButton';
+import useSearchProducts from '../../utils/hooks/useSearchProducts';
+import {
+  StyledWrapper,
+  StyledTitle,
+  StyledProductInfoWrapper,
+  StyledProductInfo,
+  StyledProductInfoTag,
+  StyledProductInfoTable,
+  StyledProductAddToCartWrapper,
+  StyledProductAddToCartInput,
+  StyledProductInfoSecondary,
+} from './styled';
+
+function ProductDetail() {
+  const { productId } = useParams();
+  const { data: product, isLoading: isLoadingProduct } = useSearchProducts({
```

Comment:
Since there is only one `loading` on the page, you can rename from `isLoadingProduct` for only `isLoading`.